### ISSUE TITLE
Conditional directives for strerror_r don't work with musl

### ISSUE THREAD

**Issue by nightlark** [gulrak/filesystem#21]
**Describe the bug**
Using musl instead of glibc results in the #else branch in `systemErrorText` getting used; musl follows the POSIX standard.

**To Reproduce**
Compiling a program using ghc::filesystem on Alpine Linux (uses musl as its libc by default) as part of a CI build is how I encountered this issue. Otherwise, setup another compiler to use musl instead of glibc.

**Expected behavior**
The variant of strerror_r that returns an int should be used.

**Additional context**
Line 208 of the log in https://cloud.drone.io/nightlark/HELICS-src/22/2/4 is where I first saw the problem.

**Comment by gulrak** [comment]
Well, musl has the XSI-compliant version of `strerror_r` wrapped with an irritating:
```
#if defined(_POSIX_SOURCE) || defined(_POSIX_C_SOURCE) \
 || defined(_XOPEN_SOURCE) || defined(_GNU_SOURCE) \
 || defined(_BSD_SOURCE)
```
which doesn't seem match the man-file canonic test of `(_POSIX_C_SOURCE >= 200112L || _XOPEN_SOURCE >= 600) && ! _GNU_SOURCE` because of the `_GNU_SOURCE` difference.
I decided to use a different approach instead of keeping to try to find a perfect set of macro tests, as I guess there will be more distributions with unusual choices in this function. Android is also special in this regard.

Still need to run a few tests, but expect something on a feature branch soon, I plan to make a "bugfix" release on the weekend.

**Comment by nightlark** [comment]
Yea, getting rid of the macro is the [change](https://github.com/nightlark/HELICS-src/commit/3832686480d271dc6941593fb10c822aec0f6341) I tried -- it took care of the compile errors with musl.

**Comment by gulrak** [comment]
Yeah, should be much more stable over the platforms, to simply dispatch. And all filesystem tests on the CI tested platforms are green, so I'll merge the change into master.

**Comment by gulrak** [comment]
Fix is now part of release v1.2.2.
